src/queryselector/parser.rs

```diff
@@ -83,7 +83,9 @@ impl<'a> Parser<'a> {
             }
             Some(b'=') => {
                 self.stream.advance();
+                self.stream.expect_and_skip(b'"')?;
                 let value = self.read_identifier();
+                self.stream.expect_and_skip(b'"')?;
```

Comment:
I think this shouldn't use `?` on the first line, as quotes aren't necessarily required. Right now it would bail out if it got `p[id=test]`. I believe quotes are only needed for when there are special characters in the value, like `:`, otherwise they are optional.
Also, instead of hardcoding `"` here, can this use `expect_oneof_and_skip(&[b'"', b'\''])`? Would be nice if this could also accept single quotes (they're valid too). 

```suggestion
                let quote = self.stream.expect_oneof_and_skip(&[b'"', b'\'']);
                let value = self.read_identifier();
                if let Some(quote) = quote {
                  // Only require the given quote if the value starts with a quote
                  self.stream.expect_and_skip(quote)?;
                }
```